api/provider/aws/test_aws/load_balancer_delete_test.go

```diff
@@ -106,8 +118,96 @@ func TestLoadBalancerDeleteIdempotence(t *testing.T) {
 		DescribeSecurityGroups(gomock.Any()).
 		Return(&ec2.DescribeSecurityGroupsOutput{}, nil)
 
+	mockAWS.EC2.EXPECT().
+		DescribeSecurityGroups(gomock.Any()).
+		Return(&ec2.DescribeSecurityGroupsOutput{}, nil)
+
+	target := provider.NewLoadBalancerProvider(mockAWS.Client(), tagStore, mockConfig)
+	if err := target.Delete("lb_id"); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestLoadBalancerDeleteRetry(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
+	tags := models.Tags{
+		{
+			EntityID:   "lb_id",
+			EntityType: "load_balancer",
+			Key:        "name",
+			Value:      "lb_name",
+		},
+		{
+			EntityID:   "lb_id",
+			EntityType: "load_balancer",
+			Key:        "environment_id",
+			Value:      "env_id",
+		},
+	}
+
+	for _, tag := range tags {
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	deleteLBInput := &elb.DeleteLoadBalancerInput{}
+	deleteLBInput.SetLoadBalancerName("l0-test-lb_id")
+
+	mockAWS.ELB.EXPECT().
+		DeleteLoadBalancer(deleteLBInput).
+		Return(&elb.DeleteLoadBalancerOutput{}, nil)
+
+	mockAWS.ELB.EXPECT().
+		DescribeLoadBalancers(gomock.Any()).
+		Return(&elb.DescribeLoadBalancersOutput{}, nil)
+
+	deleteRolePolicyInput := &iam.DeleteRolePolicyInput{}
+	deleteRolePolicyInput.SetRoleName("l0-test-lb_id-lb")
+	deleteRolePolicyInput.SetPolicyName("l0-test-lb_id-lb")
+
+	mockAWS.IAM.EXPECT().
+		DeleteRolePolicy(deleteRolePolicyInput).
+		Return(&iam.DeleteRolePolicyOutput{}, nil)
+
+	deleteRoleInput := &iam.DeleteRoleInput{}
+	deleteRoleInput.SetRoleName("l0-test-lb_id-lb")
+
+	mockAWS.IAM.EXPECT().
+		DeleteRole(deleteRoleInput).
+		Return(&iam.DeleteRoleOutput{}, nil)
+
+	readSGHelper(mockAWS, "l0-test-lb_id-lb", "lb_sg")
+	deleteSGHelper(mockAWS, "lb_sg")
+
+	securityGroup := &ec2.SecurityGroup{}
+	securityGroup.SetGroupName("l0-test-lb_id-lb")
+	securityGroup.SetGroupId("lb_sg")
+
+	output := &ec2.DescribeSecurityGroupsOutput{}
+	output.SetSecurityGroups([]*ec2.SecurityGroup{securityGroup})
+
+	gomock.InOrder(
+		mockAWS.EC2.EXPECT().
+			DescribeSecurityGroups(gomock.Any()).
+			Return(output, nil),
+		mockAWS.EC2.EXPECT().
+			DescribeSecurityGroups(gomock.Any()).
+			Return(&ec2.DescribeSecurityGroupsOutput{}, nil),
+	)
+
 	target := provider.NewLoadBalancerProvider(mockAWS.Client(), tagStore, mockConfig)
 	if err := target.Delete("lb_id"); err != nil {
 		t.Fatal(err)
 	}
+
+	assert.Len(t, tagStore.Tags(), 0)
```

Comment:
This is not the behavior we are trying to verify in this test, so this should be removed.